src/Microsoft.Health.Api/Features/HealthChecks/CachedHealthCheck.cs

```diff
@@ -10,91 +10,133 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Microsoft.Health.Core;
 
 namespace Microsoft.Health.Api.Features.HealthChecks
 {
-    public class CachedHealthCheck : IHealthCheck, IDisposable
+    internal sealed class CachedHealthCheck : IHealthCheck, IDisposable
     {
         private readonly IServiceProvider _provider;
         private readonly Func<IServiceProvider, IHealthCheck> _healthCheck;
-        private readonly ILogger<CachedHealthCheck> _logger;
-        private DateTimeOffset _lastChecked;
-        private HealthCheckResult _lastResult;
+        private readonly HealthCheckCachingOptions _options;
+        private readonly ILogger _logger;
         private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
-        private static readonly TimeSpan CacheTime = TimeSpan.FromSeconds(1);
 
-        public CachedHealthCheck(IServiceProvider provider, Func<IServiceProvider, IHealthCheck> healthCheck, ILogger<CachedHealthCheck> logger)
-        {
-            EnsureArg.IsNotNull(provider, nameof(provider));
-            EnsureArg.IsNotNull(healthCheck, nameof(healthCheck));
-            EnsureArg.IsNotNull(logger, nameof(logger));
+        // By default, the times are DateTimeOffset.MinValue such that they are always considered invalid
+        private volatile CachedHealthCheckResult _cachedResult = new CachedHealthCheckResult();
 
-            _provider = provider;
-            _healthCheck = healthCheck;
-            _logger = logger;
+        public CachedHealthCheck(IServiceProvider provider, Func<IServiceProvider, IHealthCheck> healthCheck)
+        {
+            _provider = EnsureArg.IsNotNull(provider, nameof(provider));
+            _healthCheck = EnsureArg.IsNotNull(healthCheck, nameof(healthCheck));
+            _options = provider.GetRequiredService<IOptions<HealthCheckCachingOptions>>()?.Value;
+            _logger = provider.GetRequiredService<ILogger<CachedHealthCheck>>();
```

Comment:
Just a thought, but we should try to restrict the use of this to the `PostConfigure`. I think it makes more sense to have it there since it's more of a common pattern there.

Reply:
Perhaps? We may want to keep an options object in the health check

Reply:
Move the DI to the post configure